test/Memory.hs

```diff
@@ -0,0 +1,107 @@
+module Main where
+
+import           Data.Aeson           ((.=))
+import qualified Data.Aeson           as JSON
+import           Network.Wai          (Application)
+import           SpecHelper           (posgrestTestApp, requestAction,
+                                       testCfgMemory)
+
+import           Data.ByteString.Lazy as BS hiding (take)
+import           Network.Wai.Test     (SResponse)
+import           Test.Hspec.Wai       (WaiSession, get, patch, post)
+import           Weigh                (Weigh, mainWith, maxAllocs,
+                                       validateAction, wgroup)
+
+
+import           Protolude            hiding (get)
+
+
+newtype MegaBytes = MegaBytes Double
+
+
+main :: IO ()
+main = do
+  postgrestApp <- posgrestTestApp
+  let test = testMaxAllocationMB $ postgrestApp testCfgMemory
+
+  mainWith $ do
+    test "root request" (get "/") $ MegaBytes 30
+    jsonKeyTest1M test
+    jsonKeyTest10M test
+    jsonKeyTest50M test
+    jsonArrayTest test
+
+type MemoryCaseTester = (Text -> WaiSession SResponse -> MegaBytes -> Weigh ())
+
+jsonKeyTest1M :: MemoryCaseTester -> Weigh ()
+jsonKeyTest1M test = wgroup "json key test - 1M blob" $ do
+    test "post rpc request"
+         (post "/rpc/leak?columns=blob"           $ jsonBlob 1) $
+         MegaBytes 10
+    test "post request"
+         (post "/leak?columns=blob"               $ jsonBlob 1) $
+         MegaBytes 10
+    test "patch request"
+         (patch "/leak/leak?id=eq.1&columns=blob" $ jsonBlob 1) $
```

Comment:
`/leak/leak?id=eq..` is wrong here. Should be just `/leak?id=eq..`

Reply:
Fixing this also made the patch requests consume around the same memory as the other requests resolving that question.